src/components/data-processing/ConfigureEmbedding/ConfigureEmbedding.jsx

```diff
@@ -217,17 +217,27 @@ const ConfigureEmbedding = (props) => {
     (state) => state.componentConfig[plots[selectedPlot].plotUuid]?.outstandingChanges,
   );
 
-  const config = useSelector(
-    (state) => state.componentConfig[plots[selectedPlot].plotUuid]?.config,
-  );
+  const plotConfigs = useSelector((state) => {
+    const plotUuids = Object.values(plots).map((plotIt) => plotIt.plotUuid);
+
+    const plotConfigsToReturn = plotUuids.reduce((acum, plotUuidIt) => {
+      // eslint-disable-next-line no-param-reassign
+      acum[plotUuidIt] = state.componentConfig[plotUuidIt]?.config;
+      return acum;
+    }, {});
+
+    return plotConfigsToReturn;
+  });
```

Comment:
I am not sure I understand this piece of code very well, can you explain what are you trying to achieve here?

Reply:
It selects all the plot configs that are used in this step.

Reply:
Do you mean for both frequency plot and the embedding plot?

Reply:
Before we only loaded the currently selected plot's config here and checked it to decide whether we needed to load the plot config/data for all plots.

This is not enough anymore because we are loading the same plots data in different steps, so for a particular one plot might be loaded but the rest might not be, so we should do an individual check on each plotConfig to decide whether to load it.

Reply:
Frequency plot is a different situation from embedding because it generates its own plot data so it doesn't need to get plot data from anywhere. All we need to do for it is block it until embedding is finished (so we have the updated cellSets).

Reply:
I see, that makes a lot of sense. On this one: `so we should do an individual check on each plotConfig to decide whether to load it.`  How do we do that check?

Reply:
here, added a little explanation of the change in the comment
https://github.com/biomage-ltd/ui/pull/202/files#r611712515

Reply:
sounds good, thanks for the explanation